### DISCUSSION TITLE
Adding captions for FileUpload/SpatieMediaLibraryFileUpload

### DISCUSSION THREAD

**Discussion by bilogic** [filamentphp/filament#8279] [Ideas]
1. I would like to be able to add captions after uploading files, 1 for each file
2. I have given it a shot over the past few days, but it seems that coming up with an elegant architecture is beyond me at this point
3. I'm happy to write the code if someone can advise on the architecture

**Comment by leandrocfe** [comment]
I've never tried it before, but you could take a look at [ralphjsmit-media-library-manager](https://filamentphp.com/plugins/ralphjsmit-media-library-manager)

**Reply by bilogic** [reply]
Thanks, captions was in the image gallery, but could not see it in action on a resource/page

**Reply by leandrocfe** [reply]
Could you provide more details about the issue please? What code did you try?

**Reply by bilogic** [reply]
I played with the demo here https://demo.filament-media-library.ralphjsmit.com/

~If you go under Posts and Authors, I can't add captions~

I want the images to be separate between resources, and not from a common library. Is that possible?

**Reply by leandrocfe** [reply]
You should ask on [media-library-pro](https://discordapp.com/channels/883083792112300104/961393209639067698)

**Comment by bilogic** [comment]
I made some progress on this, but it involves making quite a number of changes including some JS and `InteractsWithForms.php`. 

Questions to the project contributors, 

1. Will such a feature be accept as a PR?
2. It is a pretty big commit spanning many files, should I squash everything into 1 commit?
3. I suppose style won't be an issue since I noticed a style fixer github action? I'm using pint v1.13.1

**Reply by danharrin** [reply]
It really depends on the implementation, if its incredibly complex then I won't merge it. And adding field-specific stuff to InteractsWithForms potentially doesn't sound like the right implementation...

**Reply by bilogic** [reply]
Thanks for getting back.

It is quite big, touching many files, but I don't think it is complex. I called it `captionUploadedFile` (based off `deleteUploadedFile` found in `InteractsWithForms`). Maybe let me get it to a MVP state and let you have a look.

**Reply by danharrin** [reply]
What is the UI for adding and editing captions, is it part of Filepond?

**Reply by bilogic** [reply]
No, not part of Filepond it is based off this https://github.com/pqina/filepond/issues/208#issuecomment-1704620279. 

Frankly, the UI needs work, but its something I will get to only after it works. Right now, I'm trying to make captions show for existing images.

My WIP is here https://github.com/bilogic/filament/tree/image-caption
I'm open to any inputs you have since I'm really new to filepond and filament, it is quite a tedious trial and error doing this alone.

Thank you.

**Reply by danharrin** [reply]
You'll need to open a PR or I'll forget about doing it

**Reply by danharrin** [reply]
even a draft PR is fine

**Reply by bilogic** [reply]
Ok, I don't quite understand draft PR, what does that mean? Is it different from a normal PR? Any specific things I should take note of?

The UI is able to read and write captions using `SpatieMediaLibraryFileUpload` already and I'm more concerned about your architecture expectations before I spent any more time on this. Hope you can understand it is painful to see my efforts go down the drain sometimes.

**Reply by danharrin** [reply]
You click "submit as draft" or whatever when you open the PR. Alternatively, just open a normal PR and I can convert it to draft after

> Hope you can understand it is painful to see my efforts go down the drain sometimes.

Yeah I am aware, its just the reality of open source when maintainers are the ones who have to keep the feature alive and have responsibility for it

**Reply by scritrolf** [reply]
Did anything happen with regard to this? I would need a checkbox for each of the uploaded images and would want to save the value as custom property. Are there any pointers?
